Zend/zend_execute.c

```diff
@@ -3134,12 +3210,21 @@ ZEND_API zval* zend_assign_to_typed_ref(zval *variable_ptr, zval *value, zend_uc
 	return variable_ptr;
 }
 
-ZEND_API zend_bool ZEND_FASTCALL zend_verify_prop_assignable_by_ref(zend_property_info *prop_info, zval *orig_val, zend_bool strict) {
+ZEND_API zend_bool ZEND_FASTCALL zend_verify_prop_assignable_by_ref(zend_property_info *prop_info, zval *orig_val, zend_bool strict, zend_bool check_final) {
 	zval *val = orig_val;
+
 	if (Z_ISREF_P(val) && ZEND_REF_HAS_TYPE_SOURCES(Z_REF_P(val))) {
 		int result;
 
 		val = Z_REFVAL_P(val);
+
+		if (check_final &&prop_info && prop_info->flags & ZEND_ACC_FINAL && Z_PROP_FLAG_P(val) != IS_PROP_UNINIT) {
```

Comment:
```suggestion
		if (check_final && prop_info && prop_info->flags & ZEND_ACC_FINAL && Z_PROP_FLAG_P(val) != IS_PROP_UNINIT) {
```

Reply:
I don't think the `IS_PROP_UNINIT` check here makes sense. `val` is the value you're assigning (not the property value), so it does not support this flag (it should be uninitialized memory in fact).

Reply:
I should have noticed it :) And the check doesn't make sense because even non-initialized properties shouldn't be assigned by ref.

Then I'll do the this validation in `zend_assign_to_typed_property_reference()` and that way at least we don't need the `check_final` flag anymore.